src/main/java/org/opensearch/knn/plugin/script/KNNScoreScriptLeafFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.script;
+
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.IndexSearcher;
+import org.opensearch.knn.plugin.stats.KNNCounter;
+import org.opensearch.script.ScoreScript;
+import org.opensearch.search.lookup.SearchLookup;
+
+public class KNNScoreScriptLeafFactory implements ScoreScript.LeafFactory {
```

Comment:
Can you please add some description for this class

Reply:
Yes added.